crates/solvers/src/domain/solver.rs

```diff
@@ -193,40 +197,64 @@ impl Inner {
             let compute_solution = async |request: Request| -> Option<Solution> {
                 let wrappers = request.wrappers.clone();
                 let route = boundary_solver.route(request, self.max_hops).await?;
-                let interactions = route
-                    .segments
-                    .iter()
-                    .map(|segment| {
-                        solution::Interaction::Liquidity(Box::new(solution::LiquidityInteraction {
-                            liquidity: segment.liquidity.clone(),
-                            input: segment.input,
-                            output: segment.output,
-                            // TODO does the baseline solver know about this optimization?
-                            internalize: false,
-                        }))
-                    })
-                    .collect();
+                let interactions;
+                let gas;
+                let (input, mut output);
+
+                if !route.is_empty() {
```

Comment:
Since we probably need to have an if here anyway we can probably also leave the routing logic completely untouched (i.e. revert changes for allowing empty route segments). Rather than having the `if` after calling `route()` here we could instead have the `if sell == buy` and only call `route()` in the other case, right?